### PR TITLE
fix: Load workflows with unconnected Switch outputs

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary

Workflows from older versions of n8n with Switch nodes use `null` instead of `[]` to state that output index is not connected.

To reproduce:
1. Switch to new canvas
2. Open template at /workflows/templates/2320
3. Try to delete node

Or else
1. save workflow in 1.43.0 with a switch node not connected in first indexes 
2. update to 1.70.1
3. open workflow in new canvas, try to delete node

## Related Linear tickets, Github issues, and Community forum posts
https://github.com/n8n-io/n8n/issues/11868

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/stores/workflows.store.ts [hunk 1]
```diff
@@ -891,23 +891,28 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
...
 		// Add the new connection if it does not exist already
 		if (!connectionExists) {
-			workflow.value.connections[sourceData.node][sourceData.type][sourceData.index].push(
-				destinationData,
-			);
+			nodeConnections[sourceData.index] = nodeConnections[sourceData.index] ?? [];
+			const connections = nodeConnections[sourceData.index];
+			if (connections) {
+				connections.push(destinationData);
+			}
```
Comment (mutdmour) [inline packages/editor-ui/src/stores/workflows.store.ts:915]: feel free to suggest a neater approach here that also makes typescript happy

**Review by netroy — COMMENTED** [review]
backend changes LGTM

**Review by netroy — APPROVED** [review]

File: packages/editor-ui/src/components/InputNodeSelect.vue [hunk 2]
```diff
@@ -86,7 +86,7 @@ function getMultipleNodesText(nodeName: string): string {
 		props.workflow.connectionsByDestinationNode[activeNode.value.name].main || [];
 	// Collect indexes of connected nodes
 	const connectedInputIndexes = activeNodeConnections.reduce((acc: number[], node, index) => {
-		if (node[0] && node[0].node === nodeName) return [...acc, index];
+		if (node?.[0] && node[0].node === nodeName) return [...acc, index];
```
Comment (alexgrozav) [inline packages/editor-ui/src/components/InputNodeSelect.vue:89]:
```suggestion
		if (node?.[0]?.node === nodeName) return [...acc, index];
```

File: packages/editor-ui/src/components/CodeNodeEditor/completions/jsonField.completions.ts [hunk 3]
```diff
@@ -176,7 +176,7 @@ function useJsonFieldCompletions() {
 			if (activeNode) {
 				const workflow = workflowsStore.getCurrentWorkflow();
 				const input = workflow.connectionsByDestinationNode[activeNode.name];
-				return input.main[0][0].node;
+				return input.main[0] ? input.main[0][0].node : null;
```
Comment (alexgrozav) [inline packages/editor-ui/src/components/CodeNodeEditor/completions/jsonField.completions.ts:179]:
```suggestion
				return input.main[0]?.[0]?.node ?? null;
```

File: packages/editor-ui/src/views/NodeView.vue [hunk 4]
```diff
@@ -1394,7 +1394,11 @@ export default defineComponent({
 					lastSelectedNode.name,
 				);
 
-				if (connections.main === undefined || connections.main.length === 0) {
+				if (
+					connections.main === undefined ||
+					connections.main.length === 0 ||
+					!connections.main[0]
+				) {
```
Comment (alexgrozav) [inline packages/editor-ui/src/views/NodeView.vue:1401]:
```suggestion
				if (!connections?.main?.[0]) {
```

File: packages/editor-ui/src/views/NodeView.vue [hunk 5]
```diff
@@ -1428,7 +1432,11 @@ export default defineComponent({
 
 				const connections = workflow.connectionsByDestinationNode[lastSelectedNode.name];
 
-				if (connections.main === undefined || connections.main.length === 0) {
+				if (
+					connections.main === undefined ||
+					connections.main.length === 0 ||
+					!connections.main[0]
+				) {
```
Comment (alexgrozav) [inline packages/editor-ui/src/views/NodeView.vue:1439]:
```suggestion
				if (!connections?.main?.[0]) {
```

**Review by alexgrozav — APPROVED** [review]
Works as expected. Thank you!

**Comment by mutdmour** [comment]
@alexgrozav will merge to unblock the release. Will address the specific style issues separately

**Comment by janober** [comment]
Got released with [`n8n@1.71.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.71.0)

**Pull request merged**